server/hooks.go

```diff
@@ -1,14 +1,52 @@
 package main
 
 import (
+	"bytes"
 	"fmt"
+	"html/template"
+	"strings"
 	"time"
 
 	"github.com/mattermost/mattermost-server/v6/model"
 	"github.com/mattermost/mattermost-server/v6/plugin"
 	"github.com/mattermost/mattermost-server/v6/shared/mlog"
 )
 
+// UserHasBeenCreated is invoked after a user was created.
+func (p *Plugin) UserHasBeenCreated(c *plugin.Context, user *model.User) {
+	data := p.getGlobalMessageTemplateData(user.Id)
+	if data == nil {
+		return
+	}
+
+	for _, message := range p.getWelcomeMessages() {
+		if data.User.IsGuest() && !message.IncludeGuests {
+			continue
+		}
+
+		if len(message.GlobalWelcomeMessage) > 0 {
```

Comment:
To avoid indentation

```suggestion
		if len(message.GlobalWelcomeMessage) == 0 {
		    continue
		}
```